### PR TITLE
Fix a bug in accuracy computation and make apply_attention nicer.

### PR DESCRIPTION
**Pull request by guoyang9**
The accuracy from the computation of the acc_tracker is slightly better than that of the official evaluation method. One reason for this is the overall accuracy is not constant with the mean of all the accuracy from each sample when the number of all samples cannot be divided by batch_size.

For example, Let's suppose there are 7 samples and the batch_size is 3. The corresponding accuracy is [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.3]. Following the accuracy computation in this code, the overall accuracy is (0.1+0.1+0.3)/3=5/30=1/6. However, the correct overall accuracy should be (0.1*6+0.3)/7=9/70. This could result the former being larger than the latter.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -74,7 +74,9 @@ def run(net, loader, optimizer, tracker, train=False, prefix='', epoch=0):
             idxs.append(idx.view(-1).clone())
 
         loss_tracker.append(loss.data[0])
-        acc_tracker.append(acc.mean())
+        # acc_tracker.append(acc.mean())
```
Comment (Cyanogenoid) [inline train.py:77]: Just remove this, no need to comment it out.

**Review by Cyanogenoid — COMMENTED** [review]
Thanks! This can go the other way too, i.e. result is worse than official evaluation, but I doubt that this will visibly change results either way in practice. Good to have it for correctness anyway.

**Comment by guoyang9** [comment]
I changed the [apply_attention()](https://github.com/Cyanogenoid/pytorch-vqa/blob/0942c8bbf6bca63894400e56a6752e639481b51f/model.py#L119) as shown in this branch. I found this is easier to understand and implement.

**Review by Cyanogenoid — APPROVED** [review]
Looks good to me, thanks. The original code is a bit strange because I didn't know how exactly automatic broadcasting worked back then and F.softmax didn't have a dim parameter.
